libflux/src/semantic/walk.rs

```diff
@@ -0,0 +1,1284 @@
+            Node::StringExpr(_) => write!(f, "StringExpr"),
+            Node::IntegerLit(_) => write!(f, "IntegerLit"),
+            Node::FloatLit(_) => write!(f, "FloatLit"),
+            Node::StringLit(_) => write!(f, "StringLit"),
+            Node::DurationLit(_) => write!(f, "DurationLit"),
+            Node::UintLit(_) => write!(f, "UintLit"),
+            Node::BooleanLit(_) => write!(f, "BooleanLit"),
+            Node::DateTimeLit(_) => write!(f, "DateTimeLit"),
+            Node::RegexpLit(_) => write!(f, "RegexpLit"),
+            Node::ExprStmt(_) => write!(f, "ExprStmt"),
+            Node::OptionStmt(_) => write!(f, "OptionStmt"),
+            Node::ReturnStmt(_) => write!(f, "ReturnStmt"),
+            Node::TestStmt(_) => write!(f, "TestStmt"),
+            Node::BuiltinStmt(_) => write!(f, "BuiltinStmt"),
+            Node::Block(n) => match n {
+                Block::Variable(_, _) => write!(f, "Block::Variable"),
+                Block::Expr(_, _) => write!(f, "Block::Expr"),
+                Block::Return(expr) => write!(f, "Block::Return"),
+            },
+            Node::Property(_) => write!(f, "Property"),
+            Node::TextPart(_) => write!(f, "TextPart"),
+            Node::InterpolatedPart(_) => write!(f, "InterpolatedPart"),
+            Node::VariableAssgn(_) => write!(f, "VariableAssgn"),
+            Node::MemberAssgn(_) => write!(f, "MemberAssgn"),
+        }
+    }
+}
+impl<'a> Node<'a> {
+    pub fn loc(&self) -> &SourceLocation {
+        match self {
+            Node::Package(n) => &n.loc,
+            Node::File(n) => &n.loc,
+            Node::PackageClause(n) => &n.loc,
+            Node::ImportDeclaration(n) => &n.loc,
+            Node::Identifier(n) => &n.loc,
+            Node::IdentifierExpr(n) => &n.loc,
+            Node::ArrayExpr(n) => &n.loc,
+            Node::FunctionExpr(n) => &n.loc,
+            Node::FunctionParameter(n) => &n.loc,
+            Node::LogicalExpr(n) => &n.loc,
+            Node::ObjectExpr(n) => &n.loc,
+            Node::MemberExpr(n) => &n.loc,
+            Node::IndexExpr(n) => &n.loc,
+            Node::BinaryExpr(n) => &n.loc,
+            Node::UnaryExpr(n) => &n.loc,
+            Node::CallExpr(n) => &n.loc,
+            Node::ConditionalExpr(n) => &n.loc,
+            Node::StringExpr(n) => &n.loc,
+            Node::IntegerLit(n) => &n.loc,
+            Node::FloatLit(n) => &n.loc,
+            Node::StringLit(n) => &n.loc,
+            Node::DurationLit(n) => &n.loc,
+            Node::UintLit(n) => &n.loc,
+            Node::BooleanLit(n) => &n.loc,
+            Node::DateTimeLit(n) => &n.loc,
+            Node::RegexpLit(n) => &n.loc,
+            Node::ExprStmt(n) => &n.loc,
+            Node::OptionStmt(n) => &n.loc,
+            Node::ReturnStmt(n) => &n.loc,
+            Node::TestStmt(n) => &n.loc,
+            Node::BuiltinStmt(n) => &n.loc,
+            Node::Block(n) => n.loc(),
```

Comment:
`loc` for `Block` returns the location of the statement in the block, not the entire block. Should I comment on this maybe? 🤔

Reply:
That makes sense. It's clear from the `impl`